analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.TryCatchFinally.cs

```diff
@@ -831,6 +837,193 @@ public void TryCatch_NestedThrowWithMultipleCatchFiltersOnDifferentLevels()
+            validator.ValidateTagOrder(
+                "BeforeTry",
+                "InTry",
+                "InCatch",
+                "AfterCatch");
+
+            validator.TagStates("InCatch").Single().Should().HaveExceptionOfTypeException();
+            validator.TagStates("AfterCatch").Single().Should().HaveExceptionOfTypeException(); // ToDo: the exception should be cleared at this point
+        }
+
+        [TestMethod]
+        public void Exception_LocalReference()
+        {
+            const string code = @"
+Exception exception = null;
+var tag = ""BeforeTry"";
+try
+{
+    tag = ""InTry"";
+    throw exception;
+}
+catch
+{
+    tag = ""InCatch"";
+}
+tag = ""AfterCatch"";";
+            var validator = SETestContext.CreateCS(code).Validator;
+            validator.ValidateTagOrder(
+                "BeforeTry",
+                "InTry",
+                "InCatch",
+                "AfterCatch");
+
+            validator.TagStates("InCatch").Single().Should().HaveExceptionOfTypeException();
+            validator.TagStates("AfterCatch").Single().Should().HaveExceptionOfTypeException(); // ToDo: the exception should be cleared at this point
+        }
+
+        [TestMethod]
+        public void Exception_PropertyReference()
+        {
+            const string code = @"
+var tag = ""BeforeTry"";
+try
+{
+    tag = ""InTry"";
+    throw PropertyException;
+}
+catch
+{
+    tag = ""InCatch"";
+}
+tag = ""AfterCatch"";";
+            var validator = SETestContext.CreateCS(code).Validator;
+            validator.ValidateTagOrder(
+                "BeforeTry",
+                "InTry",
+                "InCatch",
+                "AfterCatch");
+
+            validator.TagStates("InCatch").Single().Should().HaveExceptionOfTypeException();
+            validator.TagStates("AfterCatch").Single().Should().HaveExceptionOfTypeException(); // ToDo: the exception should be cleared at this point
+        }
+
+        [TestMethod]
+        public void Exception_ParameterReference()
+        {
+            const string code = @"
+var tag = ""BeforeTry"";
+try
+{
+    tag = ""InTry"";
+    throw ex;
+}
+catch
+{
+    tag = ""InCatch"";
+}
+tag = ""AfterCatch"";";
+            var validator = SETestContext.CreateCS(code, ", Exception ex").Validator;
+            validator.ValidateTagOrder(
+                "BeforeTry",
+                "InTry",
+                "InCatch",
+                "AfterCatch");
+
+            validator.TagStates("InCatch").Single().Should().HaveExceptionOfTypeException();
+            validator.TagStates("AfterCatch").Single().Should().HaveExceptionOfTypeException(); // ToDo: the exception should be cleared at this point
+        }
+
+        [TestMethod]
+        public void Exception_ArrayElementReference()
+        {
+            const string code = @"
+Exception[] exceptions = null;
+var tag = ""BeforeTry"";
+try
+{
+    tag = ""InTry"";
+    throw exceptions[0];
```

Comment:
Not sure if this is nice or ugly 😆

Reply:
Ugly in practice and nice that we support it?